src/npc.cpp

```diff
@@ -266,16 +266,16 @@ void Npc::onCreatureAppear(Creature* creature, bool isLogin)
 	}
 }
 
-void Npc::onRemoveCreature(Creature* creature, bool isLogout)
+void Npc::onRemoveCreature(std::shared_ptr<Creature> creature, bool isLogout)
 {
 	Creature::onRemoveCreature(creature, isLogout);
 
-	if (creature == this) {
+	if (creature.get() == this) {
 		closeAllShopWindows();
 		if (npcEventHandler) {
 			npcEventHandler->onCreatureDisappear(creature);
 		}
-	} else if (Player* player = creature->getPlayer()) {
+	} else if (std::shared_ptr<Player> player = creature->getPlayer()) {
```

Comment:
Could this be `const auto`
```suggestion
	} else if (const auto player = creature->getPlayer()) {
```